x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/InMemoryNoOpCommitDirectory.java

```diff
@@ -119,7 +119,8 @@ private static void ensureMutable(String name) {
             || name.startsWith("pending_segments_")
             || name.matches("^recovery\\..*\\.segments_.*$")) == false) {
 
-            throw new IllegalArgumentException("file [" + name + "] is not mutable");
+            throw new ImmutableDirectoryException("file [" + name + "] is not mutable");
         }
     }
+
```

Comment:
NIT: drop empty line

Reply:
I prefer to have a blank line here, it simplifies merge conflicts when two commits add members to the bottom of a class.